src/test/controller/BaseController.ts

```diff
@@ -5,71 +5,97 @@ import { auth } from 'firebase-admin/lib/auth';
- * */
-export const registerTestApplication = async (
-  redisClient: RedisClient,
-  firebaseAdmin: Auth,
-  geocoder: Geocoder,
-  axiosInstance?: AxiosInstance
-): Promise<Express> => {
-  const app = new App();
-  const expressApp = express();
-  await app.registerHandlersAndRoutes(
-    expressApp,
-    redisClient,
-    firebaseAdmin,
-    geocoder,
-    axiosInstance
-  );
-  return expressApp;
-};
+export class BaseController {
+  mockRedisClient: RedisClient;
+  mockFirebaseAdmin: Auth;
+  mockGeoCoder: Geocoder;
+  mockS3: S3;
+  axiosInstance: AxiosInstance;
 
-export const connectRedisClient = async (): Promise<RedisClient> => {
-  // see https://www.npmjs.com/package/redis-mock
-  return redisMock.createClient();
-};
+  constructor() {
+    this.mockRedisClient = BaseController.createRedisMock();
+    this.mockFirebaseAdmin = BaseController.createFirebaseMock();
+    this.mockGeoCoder = BaseController.createMockNodeGeocoder();
+    this.mockS3 = BaseController.createS3Mock();
+    this.axiosInstance = axios.create();
+  }
 
-export const createFirebaseMock = (): Auth => {
-  const mockAuth = new firebaseMock.MockAuthentication();
-  const mockDatabase = new firebaseMock.MockFirebase();
-  // const mockFirestore = new firebaseMock.MockFirestore();
-  const mockStorage = new firebaseMock.MockStorage();
-  // const mockMessaging = new firebaseMock.MockMessaging();
-  const mockSdk = new firebaseMock.MockFirebaseSdk(
-    // use null if your code does not use RTDB
-    (path: any) => {
-      return path ? mockDatabase.child(path) : mockDatabase;
-    },
-    // use null if your code does not use AUTHENTICATION
-    () => {
-      return mockAuth;
-    },
-    // use null if your code does not use FIRESTORE
-    // () => {
-    //   return mockFirestore;
-    // },
-    null,
-    // use null if your code does not use STORAGE
-    () => {
-      return mockStorage;
-    },
-    // use null if your code does not use MESSAGING
-    // () => {
-    //   return mockMessaging;
-    // }
-    null
-  );
-  return mockSdk.auth();
-};
+  /**
+   * Creates an express app and registers all handlers and routes
+   * * Used for integration tests that test controllers (supertest needs an instance of express app without the port binding)
+   * */
+  registerTestApplication = async (): Promise<Express> => {
+    const app = new App();
+    const expressApp = express();
+    await app.registerHandlersAndRoutes(
+      expressApp,
+      this.mockRedisClient,
+      this.mockFirebaseAdmin,
+      this.mockGeoCoder,
+      this.mockS3,
+      this.axiosInstance
+    );
+    // cleanup resources from previous bucket if possible
+    await this.mockS3.deleteBucket({ Bucket: S3_BUCKET }).promise();
+    return expressApp;
+  };
 
-export const createMockNodeGeocoder = (): Geocoder => {
-  return nodeGeocoder({
-    provider: 'openstreetmap',
-  });
-};
+  static createRedisMock = (): RedisClient => {
```

Comment:
Maybe specify these to `public`?

Reply:
Oh the default should be `public` right?

Reply:
Hmm I think you're right, I just suggested it cause we've noted the visibility for our other files too so not sure if we want to stay consistent!

Reply:
Ah yeah we have a mix of methods specifying public or not, I usually don't but I agree we should stay consistent. I think the majority do not specify public, these are the only ones I could find. But when it comes to `private/protected` we need to specify the visibility

![image](https://user-images.githubusercontent.com/49849754/113447638-1ec94980-93af-11eb-9273-5e42ca401f21.png)